src/openrct2/peep/Peep.cpp

```diff
@@ -4641,59 +4726,67 @@ static void peep_update_fixing(sint32 steps, rct_peep * peep)
     bool progressToNextSubstate = true;
     bool firstRun               = true;
 
+    if ((peep->state == PEEP_STATE_INSPECTING) &&
+        (ride->lifecycle_flags & ( RIDE_LIFECYCLE_BREAKDOWN_PENDING | RIDE_LIFECYCLE_BROKEN_DOWN)))
+    {
+	// Ride has broken down since Mechanic was called to inspect it.
+	// Mechanic identifies the breakdown and switches to fixing it.
+        peep->state     = PEEP_STATE_FIXING;
```

Comment:
the ride->mechanic_status needs to be updated no?

Ah wait it doesn't as inspecting and fixing are the same thing in terms of `mechanic_status`